numpy/f2py/__init__.py

```diff
@@ -78,6 +80,13 @@ def compile(source,
     import tempfile
     import shlex
 
+    warnings.warn(
+        "f2py.compile() will be removed in NumPy 2.0.x.\n"
+        "Write a custom subprocess.run wrapper to f2py instead",
+        VisibleDeprecationWarning,
+        stacklevel=2,
```

Comment:
Maybe we want to include an example of this in the f2py docs.

Reply:
+1 yup, though I think that'd be better suited to #25193.